src/main/java/com/desiato/puresynth/services/AuthenticationService.java

```diff
@@ -73,31 +74,23 @@ public boolean isUserAuthenticated(Token token) {
         return true;
     }
 
-    public CustomUserDetails loadUserByToken(String token) throws InvalidTokenException {
-        Optional<Session> sessionOpt = sessionRepository.findById(token);
-        if (sessionOpt.isPresent()) {
-            Session session = sessionOpt.get();
-            User user = session.getUser();
-            if (user != null) {
-                return new CustomUserDetails(user); // Directly return CustomUserDetails
-            } else {
-                throw new InvalidTokenException("Invalid token: associated user is null");
-            }
-        } else {
-            throw new InvalidTokenException("Invalid token: token not found");
-        }
+    public Optional<CustomUserDetails> loadUserByToken(String token) {
+        return sessionRepository.findById(token)
+                .map(Session::getUserId)
+                .flatMap(userRepository::findById)
+                .map(CustomUserDetails::new);
     }
 
-    public Optional<UserDetails> authenticateByToken(Token token) throws InvalidTokenException {
-        if (token == null || token.value() == null) {
+    public Optional<CustomUserDetails> authenticateByToken(PureSynthToken pureSynthToken) throws InvalidTokenException {
```

Comment:
Why do you need this and `loadUserByToken(String token)`?

Reply:
Now that you mention that I don't think I need both. But when I created I wanted them to cover two different aspects:
- loadUserByToken(String token) was there to specifically retrieve the user details, and allowing the fetching of the user
- authenticateByToken(PureSynthToken) was there to manage the initial stages of the token-based authentication. it checked that the token was valid before proceeding with other operations.

But I realized that I do not need both